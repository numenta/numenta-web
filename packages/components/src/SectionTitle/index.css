/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.open {
  composes: b--black-40 from 'tachyons';
  composes: bg-near-white from 'tachyons';
  composes: hover-bg-light-gray from 'tachyons';
}

.title {
  composes: b--black-20 from 'tachyons';
  composes: bb from 'tachyons';
  composes: dim from 'tachyons';
  composes: hover-bg-near-white from 'tachyons';
  composes: ma0 from 'tachyons';
  composes: nowrap from 'tachyons';
  composes: overflow-hidden from 'tachyons';
  composes: pb1 from 'tachyons';
  composes: pl1 from 'tachyons';
  composes: pt2 from 'tachyons';
  composes: pointer from 'tachyons';
  composes: truncate from 'tachyons';
}

.toggle {
  composes: f3 from 'tachyons';
  composes: fl from 'tachyons';
  composes: ma0 from 'tachyons';
  composes: ph2 from 'tachyons';
  composes: pointer from 'tachyons';
  composes: relative from 'tachyons';
  color: #0277bd;
  top: 0.3rem;
}
