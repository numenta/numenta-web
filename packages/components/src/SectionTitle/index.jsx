// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import IconAngleDown from 'react-icons/lib/fa/angle-down'
import IconAngleUp from 'react-icons/lib/fa/angle-up'
import React from 'react'

import Title from '../Title'

import styles from './index.css'


const SectionTitle = ({children, clickHandle, headline, open, title}) => {
  const isHeadline = headline === true
  const isOpen = open === true && !isHeadline
  const IconAngle = isOpen ? IconAngleUp : IconAngleDown
  const titleClasses = [styles.title]
  let sectionTitle = (
    <Title headline={headline}>
      {children}
    </Title>
  )

  if (!isHeadline) {
    if (isOpen) titleClasses.push(styles.open)

    /* eslint-disable jsx-a11y/no-static-element-interactions */
    sectionTitle = (
      <nav
        className={classNames(...titleClasses)}
        onClick={clickHandle}
        title={title}
      >
        <IconAngle className={styles.toggle} />
        {sectionTitle}
      </nav>
    )
    /* eslint-enable jsx-a11y/no-static-element-interactions */
  }

  return sectionTitle
}

SectionTitle.propTypes = {
  children: React.PropTypes.node.isRequired,
  clickHandle: React.PropTypes.func.isRequired,
  headline: React.PropTypes.bool.isRequired,
  open: React.PropTypes.bool.isRequired,
  title: React.PropTypes.string,
}

SectionTitle.defaultProps = {
  headline: false,
  open: false,
}

export default SectionTitle
