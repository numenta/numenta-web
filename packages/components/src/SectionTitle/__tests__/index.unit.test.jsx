import React from 'react'
import renderer from 'react-test-renderer'

import SectionTitle from '../../SectionTitle'


describe('SectionTitle React component', () => {

  describe('Headline version', () => {
    it('Renders Headline correctly', () => {
      const component = renderer.create(
        <SectionTitle
          clickHandle={() => {}}
          headline={true}
          title="Hover Hint"
        >
          Section Title
        </SectionTitle>
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Regular version', () => {
    it('Renders Regular Opened correctly', () => {
      const component = renderer.create(
        <SectionTitle
          clickHandle={() => {}}
          headline={false}
          open={true}
          title="Hover Hint"
        >
          Section Title
        </SectionTitle>
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('Renders Regular Closed correctly', () => {
      const component = renderer.create(
        <SectionTitle
          clickHandle={() => {}}
          headline={false}
          open={false}
          title="Hover Hint"
        >
          Section Title
        </SectionTitle>
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

})
