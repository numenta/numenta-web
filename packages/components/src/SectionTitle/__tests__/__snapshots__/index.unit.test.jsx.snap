exports[`SectionTitle React component Headline version Renders Headline correctly 1`] = `
<h1
  className="headline">
  Section Title
</h1>
`;

exports[`SectionTitle React component Regular version Renders Regular Closed correctly 1`] = `
<nav
  className="title"
  onClick={[Function]}
  title="Hover Hint">
  <svg
    className="toggle"
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m31 16.4q0 0.3-0.2 0.5l-10.4 10.4q-0.3 0.3-0.5 0.3t-0.6-0.3l-10.4-10.4q-0.2-0.2-0.2-0.5t0.2-0.5l1.2-1.1q0.2-0.2 0.5-0.2t0.5 0.2l8.8 8.8 8.7-8.8q0.3-0.2 0.5-0.2t0.6 0.2l1.1 1.1q0.2 0.2 0.2 0.5z" />
    </g>
  </svg>
  <h1
    className="title">
    Section Title
  </h1>
</nav>
`;

exports[`SectionTitle React component Regular version Renders Regular Opened correctly 1`] = `
<nav
  className="title open"
  onClick={[Function]}
  title="Hover Hint">
  <svg
    className="toggle"
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m31 26.4q0 0.3-0.2 0.5l-1.1 1.2q-0.3 0.2-0.6 0.2t-0.5-0.2l-8.7-8.8-8.8 8.8q-0.2 0.2-0.5 0.2t-0.5-0.2l-1.2-1.2q-0.2-0.2-0.2-0.5t0.2-0.5l10.4-10.4q0.3-0.2 0.6-0.2t0.5 0.2l10.4 10.4q0.2 0.2 0.2 0.5z" />
    </g>
  </svg>
  <h1
    className="title">
    Section Title
  </h1>
</nav>
`;
